astrophot/models/eigen_psf.py

```diff
@@ -0,0 +1,110 @@
+import torch
+import numpy as np
+
+from .psf_model_object import PSF_Model
+from ..image import PSF_Image
+from ..utils.decorators import ignore_numpy_warnings, default_internal
+from ..utils.interpolate import interp2d
+from ._shared_methods import select_target
+from ..param import Param_Unlock, Param_SoftLimits
+from .. import AP_config
+
+__all__ = ["Eigen_PSF"]
+
+class Eigen_PSF(PSF_Model):
+    """point source model which uses multiple images as a basis for the
+    PSF as it's representation for point sources. Using bilinear
+    interpolation it will shift the PSF within a pixel to accurately
+    represent the center location of a point source. There is no
+    funcitonal form for this object type as any image can be
+    supplied. Note that as an argument to the model at construction
+    one can provide "psf" as an AstroPhot PSF_Image object. Since only
+    bilinear interpolation is performed, it is recommended to provide
+    the PSF at a higher resolution than the image if it is near the
+    nyquist sampling limit. Bilinear interpolation is very fast and
+    accurate for smooth models, so this way it is possible to do the
+    expensive interpolation before optimization and save time. Note
+    that if you do this you must provide the PSF as a PSF_Image object
+    with the correct pixelscale (essentially just divide the
+    pixelscale by the upsampling factor you used).
+
+    Args:
+      eigen_basis (tensor): This is the basis set of images used to form the eigen point source, it should be a tensor with shape (N x W x H) where N is the number of eigen images, and W/H are the dimensions of the image.
+      eigen_pixelscale (float): This is the pixelscale associated with the eigen basis images.
+    
+    Parameters:
+        flux: the total flux of the point source model, represented as the log of the total flux.
+        weights: the relative amplitude of the Eigen basis modes.
+
+    """
+
+    model_type = f"eigen {PSF_Model.model_type}"
+    parameter_specs = {
+        "flux": {"units": "log10(flux/arcsec^2)"},
+        "weights": {"units": "unitless"},
+    }
+    _parameter_order = PSF_Model._parameter_order + ("flux","weights")
+    useable = True
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if "eigen_basis" not in kwargs:
+            AP_config.ap_logger.warning("Eigen basis not supplied! Assuming psf as single basis element. Please provide Eigen basis or use Pixelated_PSF model.")
```

Comment:
"is a single basis element"?  "has a single basis element"?  "uses a single a basis element"?